src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!target || !(target instanceof Herbivore)) {
+      return;
+    }
+
+    if (!target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal !== target);
```

Comment:
The logic to remove the dead animal from the alive array is correct, but it's better to encapsulate this logic inside a method of the Animal class, such as 'checkHealth', to avoid code duplication in case you need to perform this check elsewhere.